airflow/providers/facebook/ads/hooks/ads.py

```diff
@@ -0,0 +1,150 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains Facebook Ads Reporting hooks
+"""
+import time
+from typing import Any, Dict, List, Optional
+
+from facebook_business.adobjects.adaccount import AdAccount
+from facebook_business.adobjects.adreportrun import AdReportRun
+from facebook_business.adobjects.adsinsights import AdsInsights
+from facebook_business.api import Cursor, FacebookAdsApi
+from facebook_business.exceptions import FacebookRequestError
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+JOB_COMPLETED = 'Job Completed'
+
+
+class FacebookAdsReportingHook(BaseHook):
+    """
+    Hook for the Facebook Ads API
+
+    .. seealso::
+        For more information on the Facebook Ads API, take a look at the API docs:
+        https://developers.facebook.com/docs/marketing-apis/
+
+    :param facebook_ads_conn_id: Airflow Facebook Ads connection ID
+    :type facebook_ads_conn_id: str
+    :param api_version: The version of Facebook API. Default to v6.0
+    :type api_version: str
+
+    :return: list of Facebook Ads AdsInsights object(s)
+    :rtype: list[AdsInsights]
+    """
+
+    def __init__(
+        self,
+        facebook_ads_conn_id: str = "facebook_ads_default",
+        api_version: str = "v6.0",
+    ) -> None:
+        super().__init__()
+        self.facebook_ads_conn_id = facebook_ads_conn_id
+        self.facebook_ads_config: Dict[str, Any] = {}
+        self.api_version = api_version
+        self.ad_account_id: str = ""
+        self.app_id: str = ""
+        self.app_secret: str = ""
+        self.access_token: str = ""
+        self.default_params = {
+            'level': 'ad',
+            'date_preset': 'yesterday',
+        }
+
+    def _get_service(self) -> FacebookAdsApi:
+        """ Returns Facebook Ads Client using a service account"""
+        self._get_config()
+        for key, val in self.facebook_ads_config.items():
+            setattr(self, key, val)
+
+        return FacebookAdsApi.init(app_id=self.app_id,
+                                   app_secret=self.app_secret,
+                                   access_token=self.access_token,
+                                   api_version=self.api_version)
+
+    def _get_config(self) -> None:
+        """
+        Gets Facebook ads connection from meta db and sets
+        facebook_ads_config attribute with returned config file
+        """
+        conn = self.get_connection(self.facebook_ads_conn_id)
+        if "facebook_ads_client" not in conn.extra_dejson:
+            raise AirflowException("facebook_ads_client not found in extra field")
+
+        self.facebook_ads_config = conn.extra_dejson["facebook_ads_client"]
```

Comment:
What do you think to use cached_property here?  It will cause that the person who wants to use it will not have to know in which order the methods should be performed.